mentat/command/commands/search.py

```diff
@@ -26,19 +49,58 @@ async def apply(self, *args: str) -> None:
             stream.send(str(e), color="red")
             return
 
-        for i, (feature, score) in enumerate(results, start=1):
-            label = feature.ref()
-            if label.startswith(str(session_context.cwd)):
-                label = label[len(str(session_context.cwd)) + 1 :]
+        cumulative_tokens = 0
+        for i, (feature, _) in enumerate(results, start=1):
+            prefix = "\n   "
+
+            file_name = feature.rel_path(session_context.cwd)
+            file_name = colored(file_name, "blue", attrs=["bold"])
+            file_name += colored(feature.interval_string(), "light_cyan")
+
+            tokens = feature.count_tokens(config.model)
+            cumulative_tokens += tokens
+            tokens_str = colored(f"  ({tokens} tokens)", "yellow")
+            file_name += tokens_str
+
+            name = []
             if feature.name:
-                label += f' "{feature.name}"'
-            stream.send(f"{i:3} | {score:.3f} | {label}")
+                name = feature.name.split(",")
+                name = [
+                    f"{'└' if i == len(name) - 1 else '├'}─ {colored(n, 'cyan')}"
+                    for i, n in enumerate(name)
+                ]
+
+            message = f"{str(i).ljust(3)}" + prefix.join([file_name] + name + [""])
+            stream.send(message)
             if i > 1 and i % SEARCH_RESULT_BATCH_SIZE == 0:
                 # TODO: Required to avoid circular imports, but not ideal.
-                from mentat.session_input import ask_yes_no
+                from mentat.session_input import collect_user_input
 
-                stream.send("\nShow More results? ")
-                if not await ask_yes_no(default_yes=True):
+                stream.send(
+                    "(Y/n) for more results or to exit search mode.\nResults to"
+                    ' include in context: (eg: "1 3 4" or "1-4")'
+                )
```

Comment:
is this more clear?

```suggestion
                stream.send(
                    "'Y' to show more results\n"
                    "'n' to exit search mode\n"
                    "or select specific results to add to context (e.g., '1 3 4' or '1-4'):"
                )
```

Reply:
I think I prefer the current one; more concise.